### PR TITLE
Update logstash-input-beats to 6.0.1

### PR DESCRIPTION
**Pull request by basilisk487**

### CODE REVIEW

File: proxy/src/main/java/org/logstash/beats/BeatsHandler.java [hunk 1]
```diff
@@ -23,77 +20,117 @@ public BeatsHandler(IMessageListener listener) {
...
-
-        processing.compareAndSet(false, true);
-
-        for(Message message : batch.getMessages()) {
-            logger.debug("Sending a new message for the listener, sequence: " + message.getSequence());
-            messageListener.onNewMessage(ctx, message);
-
-            if(needAck(message)) {
-                ack(ctx, message);
+        if(logger.isDebugEnabled()) {
```
Comment (sushantdewan123) [inline proxy/src/main/java/org/logstash/beats/BeatsHandler.java]: Nit: missing space

File: proxy/src/main/java/org/logstash/beats/BeatsHandler.java [hunk 2]
```diff
@@ -23,77 +20,117 @@ public BeatsHandler(IMessageListener listener) {
...
+
+                if (needAck(message)) {
+                    ack(ctx, message);
+                }
             }
+        }finally{
+            //this channel is done processing this payload, instruct the connection handler to stop sending TCP keep alive
+            ctx.channel().attr(ConnectionHandler.CHANNEL_SEND_KEEP_ALIVE).get().set(false);
+            if (logger.isDebugEnabled()) {
+                logger.debug("{}: batches pending: {}", ctx.channel().id().asShortText(),ctx.channel().attr(ConnectionHandler.CHANNEL_SEND_KEEP_ALIVE).get().get());
```
Comment (sushantdewan123) [inline proxy/src/main/java/org/logstash/beats/BeatsHandler.java]: i hate scrolling :(

File: proxy/src/main/java/org/logstash/beats/BeatsHandler.java [hunk 3]
```diff
@@ -23,77 +20,117 @@ public BeatsHandler(IMessageListener listener) {
...
+            localhost = "undefined";
+        }
 
-    private void sendKeepAlive() {
-        // If we are actually blocked on processing
-        // we can send a keep alive.
-        if(processing.get()) {
-            writeAck(context, Protocol.VERSION_2, 0);
+        String remotehost;
+        if(remote != null) {
```
Comment (sushantdewan123) [inline proxy/src/main/java/org/logstash/beats/BeatsHandler.java]: missing space

File: proxy/src/main/java/org/logstash/beats/BeatsHandler.java [hunk 4]
```diff
@@ -23,77 +20,117 @@ public BeatsHandler(IMessageListener listener) {
...
+            remotehost = remote.getAddress().getHostAddress() + ":" + remote.getPort();
+        } else{
```
Comment (sushantdewan123) [inline proxy/src/main/java/org/logstash/beats/BeatsHandler.java]: missing space

File: proxy/src/main/java/org/logstash/beats/BeatsParser.java [hunk 5]
```diff
@@ -199,17 +192,12 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
...
-                Message message = new Message(sequence, (Map) MAPPER.readValue(bytes, Object.class));
-
-                batch.addMessage(message);
-
-                if(batch.size() == batch.getBatchSize()) {
-                    logger.debug("Sending batch size: " + this.batch.size() + ", windowSize: " + batch.getBatchSize() +  " , seq: " + sequence);
-
+                logger.trace("Running: READ_JSON");
+                ((V2Batch)batch).addMessage(sequence, in, requiredBytes);
+                if(batch.isComplete()) {
```
Comment (sushantdewan123) [inline proxy/src/main/java/org/logstash/beats/BeatsParser.java]: missing space

File: proxy/src/main/java/org/logstash/beats/BeatsParser.java [hunk 6]
```diff
@@ -199,17 +192,12 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
...
+                    if(logger.isTraceEnabled()) {
```
Comment (sushantdewan123) [inline proxy/src/main/java/org/logstash/beats/BeatsParser.java]: missing space

**Comment by basilisk487** [comment]
@sushantdewan123 it's a direct copy from https://github.com/logstash-plugins/logstash-input-beats/tree/master/src/main/java/org/logstash, we're keeping the formatting exactly the same as the source, so we can easily port future changes to the original plugin.

**Pull request merged**